code/modules/projectiles/guns/flamer/flamer.dm

```diff
@@ -654,9 +663,18 @@ GLOBAL_LIST_EMPTY(flamer_particles)
 		thing.handle_flamer_fire(src, damage, delta_time)
 
 	//This has been made a simple loop, for the most part flamer_fire_act() just does return, but for specific items it'll cause other effects.
-	firelevel -= 2 //reduce the intensity by 2 per tick
+
+	firelevel -= 2 + weather_smothering_strength //reduce the intensity by 2 as default or more if in weather ---- weather_smothering_strength is set as /datum/weather_event's fire_smothering_strength
+
 	return
 
+/obj/flamer_fire/proc/update_in_weather_status()
```

Comment:
```suggestion
/obj/flamer_fire/proc/update_in_weather_status()
	SIGNAL_HANDLER
```
this annotation is required on signal handling procs, it essentially is SHOULD_NOT_SLEEP, due to the way return values are handled